lib/rotoscope.rb

```diff
@@ -1,11 +1,81 @@
 # frozen_string_literal: true
 require 'rotoscope/rotoscope'
+require 'fileutils'
+require 'tempfile'
+require 'csv'
 
 class Rotoscope
-  def self.trace(output_path, blacklist = [])
-    rs = new(output_path, blacklist)
-    rs.trace { yield rs }
-    rs.close
+  InvalidStateError = Class.new(StandardError)
+
+  def self.trace(dest, blacklist: [], flatten: false)
+    dest_file = if flatten
+      Tempfile.new("rotoscope_output")
+    elsif dest.respond_to?(:to_io)
+      dest.to_io
+    else
+      File.open(dest, 'w')
+    end
+
+    begin
+      rs = Rotoscope.new(dest_file.path, blacklist)
+      rs.trace { yield rs }
+    ensure
+      rs.close
+    end
+    rs.flatten(dest) if flatten
     rs
+  ensure
+    dest_file.close unless dest_file.nil?
+    dest_file.unlink if dest_file.is_a?(Tempfile)
+  end
+
+  def flatten(dest)
+    dest_file = dest.respond_to?(:to_io) ? dest.to_io : File.open(dest, 'w')
+    flatten_into(dest_file)
+  ensure
+    dest_file.close
```

Comment:
I wasn't sure of the IO etiquette here but it makes sense not to close something we didn't open 😄 Updating both `flatten` and `trace` not to close IO objects if given.